src/components/tx-flow/flows/UpsertRecovery/UpsertRecoveryFlowReview.tsx

```diff
@@ -35,22 +35,24 @@ export function UpsertRecoveryFlowReview({
   )!.label
 
   useEffect(() => {
-    if (!web3) {
+    if (!web3ReadOnly) {
       return
     }
 
     getRecoveryUpsertTransactions({
       ...params,
-      provider: web3,
+      provider: web3ReadOnly,
       chainId: safe.chainId,
       safeAddress,
       moduleAddress,
-    }).then((transactions) => {
-      const promise = transactions.length > 1 ? createMultiSendCallOnlyTx(transactions) : createTx(transactions[0])
-
-      promise.then(setSafeTx).catch(setSafeTxError)
     })
-  }, [guardian, moduleAddress, params, safe.chainId, safeAddress, setSafeTx, setSafeTxError, web3])
+      .then((transactions) => {
+        const promise = transactions.length > 1 ? createMultiSendCallOnlyTx(transactions) : createTx(transactions[0])
+
+        promise.then(setSafeTx).catch(setSafeTxError)
+      })
+      .catch(setSafeTxError)
```

Comment:
```suggestion
      .then((transactions) => {
        return transactions.length > 1 ? createMultiSendCallOnlyTx(transactions) : createTx(transactions[0])
      })
      .then(setSafeTx)
      .catch(setSafeTxError)
```

Reply:
Implemented in e6c3ea18cc749f858c4d410d0481849a02b8dc50.